builtin/logical/pki/path_roles.go

```diff
@@ -339,7 +339,7 @@ func (b *backend) pathRoleCreate(
 		// If they are using the system default, cap it to the role max;
 		// if it was specified on the command line, make it an error
 		if len(entry.TTL) == 0 {
-			ttl = maxTTL
+			entry.TTL = entry.MaxTTL
```

Comment:
It looks like the `ttl` and `maxTTL` vars were supposed to hold the final values to be set for `entry.TTL` and `entry.MaxTTL`, which is missed out. Similar to here, line number 318 is also not setting `entry.MaxTTL`.

So probably we can retain this line as it was before and at line 349, we can assign `entry.TTL` with `ttl`, and `entry.MaxTTL` with `maxTTL`. Thoughts?